### PR TITLE
Add jwt integration, register endpoint and fix some test cases

### PR DESCRIPTION
**Pull request by vishalgupta921**
Add jwt integration, register endpoint and fix some test cases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,11 +11,18 @@ A simple seller panel to help sellers to maintain their orders
...
 export DB_HOST=jdbc:postgresql//<db-ip>:<db-port>/seller-panel
 export DB_USER=<Your db username>
 export DB_PASS=<Your db password>
 export REDIS_PORT=<redis service port>
 export REDIS_HOST=<redis service host>
+export MAIL_HOST =<Email service host>
+export MAIL_PORT =<Email service port>
+export MAIL_USERNAME =<Email username>
+export MAIL_PASSWORD =<Email password>
+export JWT_SECRET=<Jwt secret key>
```
Comment (novicedev7291) [inline README.md:24]: These are application specific so could be optional?

File: src/main/java/com/seller/panel/controller/InvitationController.java [hunk 2]
```diff
@@ -21,10 +24,22 @@
...
+
+    @Autowired
+    private Environment env;
+
     @PostMapping(EndPointConstants.Invitation.INVITE)
     public ResponseEntity<Void> invite(@NotNull @RequestBody InvitationRequest request) {
         if(StringUtils.isBlank(request.getEmail()))
             throw exceptionHandler.getException("SP-1");
+        String token = jwtTokenUtil.generateToken(request.getEmail());
+        mailService.sendEmail(request.getEmail(), "Welcome To Seller Panel", "http://localhost:"+
```
Comment (novicedev7291) [inline src/main/java/com/seller/panel/controller/InvitationController.java]: This should be UI url, must be in configuration.

File: src/main/java/com/seller/panel/controller/InvitationController.java [hunk 3]
```diff
@@ -21,10 +24,22 @@
...
+                env.getProperty("server.port")+EndPointConstants.Registration.REGISTER+"/"+jwtTokenUtil.getJtiFromToken(token));
```
Comment (novicedev7291) [inline src/main/java/com/seller/panel/controller/InvitationController.java]: Use string buffer or formatter.

File: src/main/java/com/seller/panel/controller/RegistrationController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.constraints.NotNull;
+
+@RestController
+@RequestMapping
+public class RegistrationController {
+
+    @Autowired
+    private ExceptionHandler exceptionHandler;
```
Comment (novicedev7291) [inline src/main/java/com/seller/panel/controller/RegistrationController.java]: Create a BaseController for exception handling and extends all controller from it or use composition, remove all boilerplate logic for getting exception there.

File: src/test/java/com/seller/panel/controller/InvitationControllerTest.java [hunk 5]
```diff
@@ -1,32 +1,73 @@
...
+        verify(exceptionHandler, times(1)).getException("SP-1");
+        verifyNoMoreInteractions(exceptionHandler);
+    }
+
+    @Test
+    public void shouldInviteUser() {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+        when(jwtTokenUtil.generateToken(TestDataMaker.EMAIL1)).thenReturn(StringUtils.EMPTY);
+        when(env.getProperty("server.port")).thenReturn(TestDataMaker.PORT);
```
Comment (novicedev7291) [inline src/test/java/com/seller/panel/controller/InvitationControllerTest.java]: Why `EMAIL1` and why empty?

Comment (novicedev7291) [inline src/test/java/com/seller/panel/controller/InvitationControllerTest.java]: `server.port` string itself being used at multiple places so best suited for string constants being declared.

File: src/test/java/com/seller/panel/controller/InvitationControllerTest.java [hunk 6]
```diff
@@ -1,32 +1,73 @@
...
+        when(jwtTokenUtil.getJtiFromToken(StringUtils.EMPTY)).thenReturn(StringUtils.EMPTY);
+        doNothing().when(mailService).sendEmail(any(String.class), any(String.class), any(String.class));
+        ResponseEntity<Void> responseEntity = invitationController.invite(new InvitationRequest(TestDataMaker.EMAIL1));
+        Assertions.assertEquals(HttpStatus.CREATED.value(), responseEntity.getStatusCodeValue());
```
Comment (novicedev7291) [inline src/test/java/com/seller/panel/controller/InvitationControllerTest.java]: for assertions use `assertThat` which is more readable and for test cases, start using junit 5 features.

**Review by novicedev7291 — COMMENTED** [review]
Use `assertThat` api for all checks in test cases and do not use assertNotNull else.

File: README.md (hunk 1 above)
Comment (vishalgupta921) [reply README.md:24]: yes other than username and password all others could be optional

Comment (novicedev7291) [reply README.md:24]: I meant that jwt secret doesn't have to be in env variables.

**Review by novicedev7291 — APPROVED** [review]

**Pull request merged**